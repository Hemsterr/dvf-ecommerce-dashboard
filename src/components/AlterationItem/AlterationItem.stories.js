// Libs
import React from 'react'
import { storiesOf } from '@storybook/react'

// Components
import AlterationItem from './index'

storiesOf('AlterationItem', module).add('Default', () => (
  <AlterationItem alteration={{ label: 'Alter Shoulders/Arms' }} />
))

storiesOf('AlterationItem', module).add('Selected', () => (
  <AlterationItem
    alteration={{ label: 'Alter Shoulders/Arms', isChecked: true }}
  />
))
