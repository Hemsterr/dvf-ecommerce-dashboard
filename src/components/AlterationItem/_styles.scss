.alteration-item {
  @include text-size(pxToRem(12px), pxToRem(2px), pxToRem(25px));
  color: $black;
  font-family: $font-base-semi-bold;
  text-align: center;
  width: pxToRem(400px);
  border: pxToRem(2px) solid $big-stone;
  padding: pxToRem(23px) 0;
}
