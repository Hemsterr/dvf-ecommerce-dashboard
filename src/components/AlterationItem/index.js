// @flow
import React, { useEffect, useState } from 'react'
import ReactTooltip from 'react-tooltip'

// Components
import CheckBox from '../CheckBox'
import Input from '../Input'

// Helpers
import { getWindowDimensions } from '../../helpers/utilities'

type Props = {
  label: string,
  id: string,
  garmentId: string,
  measurementError?: string,
  isChecked: boolean,
  isInHome: boolean,
  price: number,
  handleSelectAlteration: Function,
  handleUpdateMeasurement: Function,
}

const AlterationItem = (props: Props) => {
  const {
    label,
    isChecked,
    isInHome,
    price,
    id,
    garmentId,
    measurementError,
    handleUpdateMeasurement,
    handleSelectAlteration,
  } = props

  const [windowDimensions, setWindowDimensions] = useState(
    getWindowDimensions()
  )

  useEffect(() => {
    function handleResize() {
      setWindowDimensions(getWindowDimensions())
    }
    window.addEventListener('resize', handleResize)
    return () => window.removeEventListener('resize', handleResize)
  }, [])

  const tooltipPlace = windowDimensions.width > 1200 ? 'right' : 'bottom'

  return (
    <>
      <div className="alteration__item">
        <div className="alteration__checkbox-wrapper">
          <CheckBox
            label={label}
            isChecked={isChecked}
            setChecked={() => handleSelectAlteration({ garmentId, id })}
            disabled={isInHome}
          />
          {isInHome && (
            <span
              data-tip=""
              data-for="in-home"
              className="alteration__in-home"
            />
          )}
        </div>
        <p className="alteration__price">{`${price}$`}</p>
        <ReactTooltip type="light" place={tooltipPlace} id="in-home" border>
          <div className="landing__tooltip">
            <p>available for in-home fitting only</p>
          </div>
        </ReactTooltip>
      </div>
      {isChecked && (
        <Input
          className="alteration__measurement col-xs-12 col-sm-12"
          handleOnBlur={handleUpdateMeasurement}
          errorMessage={measurementError}
          placeholder="measurement - # off"
          isMeasurement
          id={id}
          tooltipPlace={tooltipPlace}
        />
      )}
    </>
  )
}

AlterationItem.defaultProps = {
  label: '',
  id: '',
  garmentId: '',
  measurementError: '',
  isChecked: false,
  isInHome: false,
  price: 0,
  handleSelectAlteration: () => {},
  handleUpdateMeasurement: () => {},
}

const isEqualProps = (prevProps, nextProps) => prevProps.label === nextProps.label
  && prevProps.label === nextProps.label
  && prevProps.id === nextProps.id
  && prevProps.garmentId === nextProps.garmentId
  && prevProps.measurementError === nextProps.measurementError
  && prevProps.isChecked === nextProps.isChecked
  && prevProps.isInHome === nextProps.isInHome
  && prevProps.price === nextProps.price

export default React.memo<Props>(AlterationItem, (prevProps, nextProps) => isEqualProps(prevProps, nextProps)
)
