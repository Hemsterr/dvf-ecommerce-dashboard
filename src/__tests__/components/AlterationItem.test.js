// Libs
import React from 'react'
import { shallowToJson } from 'enzyme-to-json'
import { shallow } from 'enzyme'

// Component
import AlterationItem from '../../components/AlterationItem'

const mockProps = {
  alteration: 'Ciara Satin Crepe Pants',
}

const wrapper = (props = {}) => {
  const component = shallow(<AlterationItem {...props} />)

  return component
}

describe('AlterationItem component', () => {
  // Snapshot
  test('should render correctly AlterationItem component default props', () => {
    const wrapperWithoutProps = wrapper()
    expect(shallowToJson(wrapperWithoutProps)).toMatchSnapshot()
    expect(AlterationItem.defaultProps.alteration).toEqual('')
  })

  // Testing props
  test('should be rendered with props', () => {
    const wrapperWithProps = wrapper(mockProps)
    expect(shallowToJson(wrapperWithProps)).toMatchSnapshot()
    expect(
      wrapperWithProps
        .find('p')
        .last()
        .props().children
    ).toMatch(mockProps.alteration)
  })
})
