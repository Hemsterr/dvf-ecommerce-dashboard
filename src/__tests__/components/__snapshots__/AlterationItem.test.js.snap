// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlterationItem component should be rendered with props 1`] = `
<div
  className="alteration-item"
>
  <p>
    Item:
  </p>
  <p>
    Ciara Satin Crepe Pants
  </p>
</div>
`;

exports[`AlterationItem component should render correctly AlterationItem component default props 1`] = `
<div
  className="alteration-item"
>
  <p>
    Item:
  </p>
  <p />
</div>
`;
